azure-iot-device/azure/iot/device/common/pipeline/pipeline_events_base.py

```diff
@@ -50,17 +50,17 @@ class ResponseEvent(PipelineEvent):
     :type status_code: int
     :ivar request_id: The request ID which will eventually be used to match a RequestOperation
       operation to this event.
-    :type request: str
+    :type request_id: str
     :ivar response_body: The body of the response.
-    :type request_body: str
-
-    :ivar status_code:
-    :type status: int
-    :ivar respons_body:
+    :type response_body: str
+    :ivar retry_after: A retry interval value that was extracted from the topic.
+    :type retry_after: int
     """
 
-    def __init__(self, request_id, status_code, response_body):
+    def __init__(self, request_id, status_code, response_body, retry_after=None):
         super(ResponseEvent, self).__init__()
         self.request_id = request_id
         self.status_code = status_code
         self.response_body = response_body
+        # TODO Question : If not here we have to use DPSResponse
```

Comment:
># TODO Question : If not here we have to use DPSResponse [](start = 8, length = 56)

This is OK here.  The concept of "retry_after" is something that exists outside of DPS, so we may need to use it in other situations